### PR TITLE
Increase maximum clipboard buffer to 250KB

### LINKED ISSUE #9296: Add a warning about the inter-qube clipboard maximum character limit
### The problem you're addressing (if any)
The inter-qube clipboard system appears to have a 65000 character limit, which means that any text longer than this limit that is copied between individual qubes by using the ctrl+shift+c and ctrl+shift+v shortcuts will be silently truncated to 65000 characters. I noticed this in Q4.2 while trying to copy over some config files from one GUI text editor to another in a different qube, and then some syntax errors appeared in the program log. Upon inspection, I noticed that a portion of the config file in the new qube was missing.

### The solution you'd like
Please add some sort of warning message on the screen. When a user tries to copy a text section that is longer than this limit then a message should pop up notifying the user that not all text has been copied over.

### The value to a user, and who that user might be
Any user who copies large portions of text won't wonder why some of it has disappeared.

### Completion criteria checklist

(This section is for developer use only. Please do not modify it.)

### PR DESCRIPTION
**Pull request by alimirjamali**
Also making the clipboard buffer size to be user configurable on the GUI daemon side (xside).

Fixes: https://github.com/QubesOS/qubes-issues/issues/9296

### CODE REVIEW

File: version [hunk 1]
```diff
@@ -1 +1 @@
-4.2.5
+4.2.6
```
Comment (marmarek) [inline version]: Please keep the version file unchanged (I prefer done it in a separate commit, on the main branch already - otherwise it will conflict if any other release happen in the meantime)

File: include/qubes-gui-protocol.h [hunk 2]
```diff
@@ -55,6 +55,11 @@
  *  - gui-daemon sends MSG_WINDOW_DUMP_ACK to acknowledge finishing
  *    MSG_WINDOW_DUMP processing; this allows agent to know when it
  *    can safely unmap a window's grants
+ * 1.8:
+ *  - agent could send up to 256KB of clipboard data. And if clipboard is over
+ *    the limit, it will send one byte over it to trigger rejection on xside.
```
Comment (marmarek) [inline include/qubes-gui-protocol.h]: Can you clarify that sending 256KB+1 should not result in connection termination anymore? The current "rejection on xside" is `exit(1)`...

**Review by marmarek — CHANGES_REQUESTED** [review]

File: version (hunk 1 above)
Comment (alimirjamali) [reply version]: Sure. will do that.

File: include/qubes-gui-protocol.h (hunk 2 above)
Comment (alimirjamali) [reply include/qubes-gui-protocol.h]: > The current "rejection on xside" is `exit(1)`...

I deleted that exit(1). It is red in color. [Line 1025](https://github.com/QubesOS/qubes-gui-daemon/pull/150/files#diff-7897f6116dd3815ad17499205f816d770805aca6ce63e96ff55be4fe2d95a3d6L1025)

File: include/qubes-gui-protocol.h [hunk 3]
```diff
@@ -82,8 +87,11 @@ typedef unsigned __int32 uint32_t;
 /* Minimum version for MSG_WINDOW_DUMP_ACK */
 #define QUBES_GUID_MIN_MSG_WINDOW_DUMP_ACK 0x00010007
 
+/* Minimum version for (up to) 4x sized clipboard */
+#define QUBES_GUID_MIN_CLIPBOARD_4X 0x00010008
+
 //arbitrary
-#define MAX_CLIPBOARD_SIZE 65000
+#define MAX_CLIPBOARD_SIZE 256000
```
Comment (marmarek) [inline include/qubes-gui-protocol.h]: Since the old size is relevant for old protocol support, better to keep MAX_CLIPBOARD_SIZE unchanged and have new constant for increased size. As you can see in the other PR, not having this constant here forces you to open code it in a few places.

As for why keeping old one unchanged - it avoids mixing protocol versions (old agent built with new header file).

**Review by marmarek — CHANGES_REQUESTED** [review]

File: include/qubes-gui-protocol.h (hunk 3 above)
Comment (alimirjamali) [reply include/qubes-gui-protocol.h]: This one is resolved

**Comment by alimirjamali** [comment]
@marmarek Now that I finished work on the other issue (`skip-update`), I can work full force on this one (clipboard).

This is a pretty complex situation which involves multiple repositories, some of which depend on the other (specifically this one). I believe it would be better to work in stages and only go forward after overall change to previous stage is agreed. After all of them are finished, they can be approved together.

I would like to work in the following order if possible:

1. `qubes-gui-common` for the main header file and definitions
2. `qubes-gui-agent-linux` for the small vmside patch
3. `qubes-core-admin-client` for the `qubes-guid` command line option
4. `qubes-gui-daemon` major work
5. `qubes-desktop-linux-manager` the systray widget patch

Please let me know if this OK.

**Comment by marmarek** [comment]
> * `qubes-core-admin-client` for the `qubes-guid` command line option
> 
> * `qubes-gui-daemon` major work

gui-daemon should come before core-admin-client

besides that looks okay

**Comment by alimirjamali** [comment]
Personal notes:

CI/CD fails since it pulls `qubes-windows-tools-cross` for building. Which tries to download `dotNetFx40_Full_x86_x64.exe` from a dead link on archive.org. The original link from Microsoft still works, the problem is irrelevant to the PR.

**Review by marmarek — APPROVED** [review]
